### PR TITLE
Solution to dupes and missing rows on repeat tables

### PR DESCRIPTION
**Pull request by ggalmazor**
This PR solves #134

 Warning the user about missing/dupes in repeat tables would be handled in #135 

To test this PR, you can use this form: [Untitled Form.xml.tar.gz](https://github.com/opendatakit/aggregate/files/1476411/Untitled.Form.xml.tar.gz)

Just send a submission and alter the database to produce missing rows and duped rows. This example has a dupe row for `_ORDINAL_NUMBER` 1 and has no row for `_ORDINAL_NUMBER` 2.

_URI | _CREATOR_URI_USER | _CREATION_DATE | _LAST_UPDATE_URI_USER | _LAST_UPDATE_DATE | _PARENT_AURI | _ORDINAL_NUMBER | _TOP_LEVEL_AURI | SOME_TEXT
-- | -- | -- | -- | -- | -- | -- | -- | --
uuid:46666a40-dfbf-4255-bcd4-e34b33325000 | anonymousUser | 2017-11-13 22:16:31.859000 | NULL | 2017-11-15 22:16:31.859000 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 1 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 1
uuid:46666a40-dfbf-4255-bcd4-e34b33325001 | anonymousUser | 2017-11-14 22:16:31.859000 | NULL | 2017-11-15 22:16:31.859000 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 1 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 2
uuid:46666a40-dfbf-4255-bcd4-e34b33325002 | anonymousUser | 2017-11-15 22:16:31.859000 | NULL | 2017-11-15 22:16:31.859000 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 3 | uuid:f99e6bd0-e07a-4b1b-b121-a34b8c852319 | 3

With this data, you should be able to see and export a group with rows for the text '2' and '3':

demo_repeat_group

### CODE REVIEW

File: src/main/java/org/opendatakit/aggregate/submission/type/RepeatSubmissionType.java [hunk 1]
```diff
@@ -141,24 +133,33 @@ public void getValueFromEntity(CallingContext cc) throws ODKDatastoreException {
...
-      if ( ordinal == null || ordinal.longValue() != expectedOrdinal ) {
-        String errString = "SELECT * FROM " + d.getTableName()
-            + " WHERE _TOP_LEVEL_AURI = " + d.getTopLevelAuri()
-            + " AND _PARENT_AURI = " + d.getParentAuri() + " is missing a (repeat) group instance OR has an extra copies.";
-        throw new ODKEnumeratedElementException(errString);
-      }
-      ++expectedOrdinal;
-      SubmissionSet set = new SubmissionSet(enclosingSet, d, repeatGroup, form, cc);
-      submissionSets.add(set);
+    long maxOrdinalNumber = 0;
```
Comment (lognaturel) [inline src/main/java/org/opendatakit/aggregate/submission/type/RepeatSubmissionType.java]: I'd tend to mark this as `final` but that's a small thing.

Comment (lognaturel) [reply src/main/java/org/opendatakit/aggregate/submission/type/RepeatSubmissionType.java]: Would also be good to have a comment above this code block. I think the original one is still relevant?

**Comment by lognaturel** [comment]
[nested-repeats.xml.txt](https://github.com/opendatakit/aggregate/files/1497600/nested-repeats.xml.txt)

The XLSForm is [here](https://docs.google.com/spreadsheets/d/19Qp42dLkMCMj5v4_hiojtfU9UKOrVwG74RJ3V3TAX0w/edit?usp=sharing) so you can start getting used to that standard! That's how most people build forms these days.

**Comment by ggalmazor** [comment]
Let's continue with this changeset in #153. I've solved the mess I produced last day with that git rebase...

**Pull request closed without merging**